build/dsc/templates/dsc_type.rb.erb

```diff
@@ -71,7 +84,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
 <%    end -%>
     validate do |value|
 <%    if property.name.downcase == 'ensure' -%>
-      resource[:ensure] = value.downcase
+      resource[:ensure] = provider.munge_ensure(value.downcase)
```

Comment:
The `downcase` is not necessary here because `munge_ensure` is a case insensitive regex.  (It probably makes sense to handle any manipulation in `munge_ensure` as well)